src/WindowsTimezone.php

```diff
@@ -0,0 +1,114 @@
+        'Transitional Islamic State of Afghanistan Standard Time' => 'Asia/Kabul',
+        'Ekaterinburg Standard Time'                => 'Asia/Yekaterinburg',
+        'West Asia Standard Time'                   => 'Asia/Tashkent',
+        'India Standard Time'                       => 'Asia/Colombo',
+        'Nepal Standard Time'                       => 'Asia/Colombo', //fallback on India Standard Time because can be absent on outdated database
+        'Central Asia Standard Time'                => 'Asia/Almaty',
+        'Sri Lanka Standard Time'                   => 'Asia/Colombo',
+        'N. Central Asia Standard Time'             => 'Asia/Novosibirsk',
+        'Myanmar Standard Time'                     => 'Asia/Rangoon',
+        'S.E. Asia Standard Time'                   => 'Asia/Bangkok',
+        'North Asia Standard Time'                  => 'Asia/Krasnoyarsk',
+        'China Standard Time'                       => 'Asia/Shanghai',
+        'Singapore Standard Time'                   => 'Asia/Singapore',
+        'Taipei Standard Time'                      => 'Asia/Taipei',
+        'W. Australia Standard Time'                => 'Australia/Perth',
+        'North Asia East Standard Time'             => 'Asia/Irkutsk',
+        'Korea Standard Time'                       => 'Asia/Seoul',
+        'Tokyo Standard Time'                       => 'Asia/Tokyo',
+        'Yakutsk Standard Time'                     => 'Asia/Yakutsk',
+        'A.U.S. Central Standard Time'              => 'Australia/Darwin',
+        'Cen. Australia Standard Time'              => 'Australia/Adelaide',
+        'A.U.S. Eastern Standard Time'              => 'Australia/Sydney',
+        'E. Australia Standard Time'                => 'Australia/Brisbane',
+        'Tasmania Standard Time'                    => 'Australia/Hobart',
+        'Vladivostok Standard Time'                 => 'Asia/Vladivostok',
+        'West Pacific Standard Time'                => 'Pacific/Port_Moresby',
+        'Central Pacific Standard Time'             => 'Pacific/Guadalcanal',
+        'Fiji Islands Standard Time'                => 'Pacific/Fiji',
+        'New Zealand Standard Time'                 => 'Pacific/Auckland',
+        'Tonga Standard Time'                       => 'Pacific/Tongatapu'
+    ];
+
+    /**
+     * return a valid IANA timezone
+     *
+     * @param string $windowsTimezone timezone in widows format
+     *
+     * @return string
+     */
+    public function getTimezone($windowsTimezone)
+    {
+        if (!isset(self::$timezone[$windowsTimezone])) {
+            throw new TimezoneNotFoundException(sprintf('windows timezone %s is not registered', $windowsTimezone));
+        }
+
+        if (!in_array(self::$timezone[$windowsTimezone], \DateTimeZone::listIdentifiers())) {
+            throw new InvalidTimezoneException(sprintf("the timezone %s does not exists in your installation", self::$timezone[$windowsTimezone]));
+        }
```

Comment:
We could also stack transformers such as those, and chain transform the timezone, so *yes*, this could be the scope if this class. It is already transform.... We ignore it. That's it. If it is **not** *either* a windows timezone *OR* a valid timezone anyway, THEN it is invalid.

Nothing prevent microsoft from having correct timezones without saying anything (it is microsoft after all...) !